Services/src/CSClassroom/CSClassroom.WebApp/Views/Submission/Download.cshtml

```diff
@@ -0,0 +1,329 @@
+                                    @:if (numStudentsAppended > 0)
+                                    @:{
+                                        @:studentSummaryDisplay += "; ";
+                                    @:}
+                                    @:studentSummaryDisplay += "@Model.SectionsAndStudents[iSect].SelectedStudents[iStudent].LastName";
+                                    @:studentSummaryDisplay += ", ";
+                                    @:studentSummaryDisplay += "@Model.SectionsAndStudents[iSect].SelectedStudents[iStudent].FirstName";
+                                    @:numStudentsAppended++;
+                                @:}
+                            @:}
+                            @:else    @* !ctlStudentSelected.prop("checked") *@
+                            @:{
+                                @:downloadAllFromSection = false;
+                            @:}
+                        @:}
+                    @:}
+                    }  // server-side: for (int iStudent = 0...
+
+                    // Done iterating through users of this section; construct the final link text
+                    @:var remaining = numStudentsSelected - c_maxStudentsToDisplayInSelectionLink;
+                    @:if (downloadAllFromSection)
+                    @:{
+                        @:studentSummaryDisplay = "Students: All";
+                        @:if (!includeUnsubmitted)
+                        @:{
+                            @:studentSummaryDisplay += " (except unsubmitted)";
+                        @:}
+                    @:}
+                    @:else if (numStudentsAppended == 0)
+                    @:{
+                        @:studentSummaryDisplay = "Students: None";
+                    @:}
+                    @:else if (remaining > 0)
+                    @:{
+                        @:studentSummaryDisplay += "; and " + remaining + " more";
+                    @:}
+                    @:studentsSubmitButton.prop("value", studentSummaryDisplay);
+                @:}
+            @:}
+        }
+
+        @* Disable main download form submit button if no students are selected *@
+        $("#@Html.IdFor(model => model.DownloadSubmitButton)").prop("disabled", numStudentsSelectedAcrossAllSections == 0);
+    }
+
+    @* Called when user clicks the Select All or Select None button from the Student
+       Selection Form *@
+    function onSelectAllOrNone(isSelectAll)
+    {
+        @if (iSectionClicked != -1)
+        {
+            for (int iStudent = 0; iStudent < Model.SectionsAndStudents[iSectionClicked].SelectedStudents.Count; iStudent++)
+            {
+                @:var idStudentSelected = "@Html.IdFor(model => model.SectionsAndStudents[iSectionClicked].SelectedStudents[iStudent].Selected)";
+                @:var ctlStudentSelected = $("#" + idStudentSelected);
+                @:ctlStudentSelected.prop("checked", isSelectAll);
+            }
+        }
+    }
+
+</script>
```

Comment:
right now, the "show students" part appears for a split second, and then disappears. maybe it can be hidden before it is rendered (display = none)?

Reply:
Done!  omGGGG I can't believe I never noticed this before.  I was so ready to blame your fast/slow machine or something, but it's so OBVIOUS I can't unsee it now.  Adding hidden="hidden" to the divs did the trick.